Add grouping of links to markdown renderer

DigiLive opened the pull request:
To satisfy the markdown linter, links are now embedded as
reference-style links instead of standard links.
Also, the line widths of the commit titles are wrapped to 80 chars max.

DigiLive committed:
Add grouping of links to markdown renderer

To satisfy the markdown linter, links are now embedded as
reference-style links instead of standard links.
Also, the line widths of the commit titles are wrapped to 80 chars max.

DigiLive commented:
When wrapping an item of a list in a markdown file, all items of this list should be separated by a blank line.

Word-wrapping is extracted to a method, which checks if a string has been changed after wrapping.
If so, search for list-item syntax of each line and add a new line in front of it.

Should be the final changes for this PR.

JBlond reviewed (approved):
It looks fine, but you have to resolve the merge conflict.

DigiLive committed:
Merge branch 'main' into add-link-grouping

# Conflicts:
#	src/Renderers/MarkDown.php

DigiLive committed:
Fix PHPUnit tests

The content of the generated changelog is changed because of pull
request #16.
The tests needed to be updated according to these changes.

DigiLive committed:
Fix not clearing links before build

Method Markdown::build() can be called multiple times.
At building, it might collect links which weren't cleared at consecutive
calls.
